### DISCUSSION TITLE
Add text with backround inside jpg

### DISCUSSION THREAD

**Discussion by ygueparm** [ImageMagick/ImageMagick#6893] [Help] [answered]
### ImageMagick version

6.9.10-23 Q16

### Operating system, version and so on

4.4.0-22621-Microsoft (WSL debian buster)

### Description

it was a bit strange, maybe i miss something, i am very sorry if it's the case

i need to add a text in a picture with a backround around the text and a datamatrix inside the original.
first i did this and it works pretty well without backround :
```
convert original.jpg datamatrix.png -composite -strokewidth 2 -stroke black -font Helvetica-Narrow-Bold -fill "#58FA82" -pointsize 150 -draw "text 100,200 'rainbow - blue'" final.jpg
```
if i add this options because i need a backround around my text
```
-background "#ecffaa"
```
it doesn't add backround but ad only the text "rainbow - blue" like the first line :(

i change my approach first without the datamatrix with the use of Label : 
```
convert original.jpg -font Helvetica-Narrow-Bold  -strokewidth 2 -stroke black  -fill "#58FA82" -pointsize 150 label:'Rainbow - blue' final.jpg
```
but now i have 2 files : final-0.jpg  final-1.jpg  one with my texte and the other with original without my text.

do you think it would be possible to have everything in the same command line to have the background and text on the original image? for the datamatrix, I can do without it for the moment.

**Comment by EPPI-c** [comment, marked as answer]
```sh
convert original.jpg datamatrix.png -composite -undercolor "#ecffaa" -strokewidth 2 -stroke black -font Helvetica-Narrow-Bold -fill "#58FA82" -pointsize 50 -annotate +100+200 "rainbow - blue" final.jpg
```
like this?

**Reply by ygueparm** [reply]
my bad, it works exactly the way i want it to. i'm ashamed it looks so simple. 😭 I got stuck on the background translation (french-> english) without thinking of the undercolor option.
